api/apps/api/src/modules/scenarios/project-checker/project-checker.service-real.ts

```diff
@@ -8,19 +8,46 @@ import { Project } from '@marxan-api/modules/projects/project.api.entity';
 import { InjectRepository } from '@nestjs/typeorm';
 import { PlanningAreasService } from '@marxan-api/modules/planning-areas/planning-areas.service';
 import { isDefined } from '@marxan/utils';
-
-export const doesntExist = Symbol(`doesn't exist`);
-export type DoesntExist = typeof doesntExist;
+import {
+  doesntExist,
+  DoesntExist,
+  hasPendingExport,
+  HasPendingExport,
+  ProjectChecker,
+} from '@marxan-api/modules/scenarios/project-checker/project-checker.service';
 
 @Injectable()
-export class ProjectChecker {
+export class ProjectCheckerReal implements ProjectChecker {
   constructor(
     private readonly apiEvents: ApiEventsService,
     @InjectRepository(Project)
     private readonly repository: Repository<Project>,
     private readonly planningAreas: PlanningAreasService,
   ) {}
 
+  async hasProjectPendingExports(
```

Comment:
minor, I suggest to keep this to an actual question word order such as

```suggestion
  async projectHasPendingExports(
```
or even
```suggestion
  async hasPendingExports(
```

Reply:
Much clearer!